main/menu/menu_utils.c

```diff
@@ -0,0 +1,132 @@
+#include "menu_utils.h"
+
+#include <stdio.h>
+#include <stdbool.h>
+
+/**
+ * @brief Write and return the proper text label to use for the given menu item
+ *
+ * If the menu item's label is a simple string, it will be returned directly.
+ * If the menu item's label must be constructed from multiple strings, it
+ * will be written to \c buffer, and a pointer to \c buffer will be returned.
+ *
+ * @param buffer   A buffer to use for constructing the label, if needed
+ * @param buflen   The size of the buffer
+ * @param item     The menu item whose label to return
+ *
+ * @return A pointer to the item's label text
+ */
+const char* getMenuItemLabelText(char* buffer, int buflen, const menuItem_t* item)
+{
+    if (item->minSetting != item->maxSetting)
+    {
+        // Handle concatenating the values
+        if (item->options)
+        {
+            snprintf(buffer, buflen - 1, "%s%s", item->label, item->options[item->currentOpt]);
+        }
+        else
+        {
+            snprintf(buffer, buflen - 1, "%s: %d", item->label, item->currentSetting);
+        }
+
+        // Return the buffer passed to us
+        return buffer;
+    }
+    else if (item->label)
+    {
+        // Just return the item label
+        return item->label;
+    }
+    else if (item->options)
+    {
+        // Just return the option label
+        return item->options[item->currentOpt];
+    }
+    else
+    {
+        // Return an empty string
+        buffer[0] = '\0';
+        return buffer;
+    }
+}
+
+/**
+ * @brief Returns whether a menu item is a settings item
+ *
+ * This will return true for items that were added with either ::addSettingsItemToMenu()
+ * or ::addSettingsOptionsItemToMenu()
```

Comment:
You don't need the `::` for functions with parenthesis. Doxygen sees `()` and does the right thing. You do need `::` for non-function things you want to reference. No harm, no foul though